### PR TITLE
[Feature] Custom title bar

### PR DESCRIPTION
**Pull request by Nocccer**
Via the pacakge https://www.npmjs.com/package/custom-electron-titlebar it is possible and easy to add a custom titlebar and get rid of the os provided one.

- Added support for font and theme
https://user-images.githubusercontent.com/61798668/210848893-e2763294-b7ba-4ecf-bde9-da90965cbb17.mp4
- Titlebar supports electron dev menu in yarn start
image

Todo:
- Add support for f11 again. Only working in development mode somehow.
- Add custom icons for minimize, maximize and close.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/main_window.ts [hunk 1]
```diff
@@ -66,12 +70,16 @@ export const createMainWindow = () => {
     }
   }
 
+  // setup the titlebar main process
+  setupTitlebar()
```
Comment (arielj) [inline src/backend/main_window.ts:74]: I think it's important to make this optional

- with this custom titlebar we lose the features the window manager shows when you right click the title bar for example
- users may have their own window decoration style/personalizations they want to use instead (like text alignment, font, icons, etc)

**Comment by flavioislima** [comment]
Tested here and it is not working on macOS:
image

**Comment by Nocccer** [comment]
> Tested here and it is not working on macOS: image

Seems that only the button does not work.

**Comment by Nocccer** [comment]
Closed because macos is not well supported.

**Comment by flavioislima** [comment]
I think should be fine to skip macOS on that and support windows and linux only,

**Comment by imLinguin** [comment]
On Mac we could add this without window handles, since we can still position the trafficLights and make it integrate nicely into the UI

**Comment by Nocccer** [comment]
> On Mac we could add this without window handles, since we can still position the trafficLight and make it integrate nicely into the UI
> 
> https://www.electronjs.org/docs/latest/tutorial/window-customization#customize-the-traffic-light-position-macos

We could do that for all os like that.

**Comment by imLinguin** [comment]
Yeah, I bet that's what that lib is doing under the hood

**Comment by Nocccer** [comment]
> Yeah, I bet that's what that lib is doing under the hood

Maybe. It could also do more. So maybe it is easier to write our own solution if we want that.

**Comment by imLinguin** [comment]
We can achieve that with Electron's magic https://www.electronjs.org/docs/latest/tutorial/window-customization#set-custom-draggable-region

**Comment by Nocccer** [comment]
> We can achieve that with Electron's magic https://www.electronjs.org/docs/latest/tutorial/window-customization#set-custom-draggable-region

Nice works great. Even double click maximize it.

**Pull request closed without merging**